spec/models/user_spec.rb

```diff
@@ -149,4 +149,44 @@
       expect(user.name).to eq 'Axe'
     end
   end
+
+  describe 'teams_data should return a list of teams' do
+    #  needs persistence because is testing a sql query
+    let(:team) { create :team }
+    let(:user_one) do
+      create :user, email: 'user1@asdf.com', current_team_id: team.id
+    end
+    let(:user_two) { create :user, email: 'user2@asdf.com' }
+
+    it 'in a specific format' do
+      create :user_team, team: team, user: user_one
+      expected_result = {
+        id: team.id,
+        name: team.name,
+        members: 1,
+        role: 2,
+        current_team: true
+      }
+
+      user_one.teams_data.first.each do |k, v|
+        expect(v).to eq(expected_result.fetch(k.to_sym))
+      end
+    end
+
+    it 'should return correct number of team members' do
+      create :user_team, team: team, user: user_one
+      create :user_team, team: team, user: user_two
+      expected_result = {
+        id: team.id,
+        name: team.name,
+        members: 2,
+        role: 2,
+        current_team: true
+      }
+
+      user_one.teams_data.first.each do |k, v|
+        expect(v).to eq(expected_result.fetch(k.to_sym))
+      end
+    end
+  end
```

Comment:
There should be one more test case

3 teams (or 2) and one user does not belong to this team